### PR TITLE
Support new backend API for key entities

### PR DESCRIPTION
**Pull request by allanlang**
Supports (but does not complete!) #63 through the following:

- Addition of backend services integrating with Protect Earth API, including all model objects for serialization
- Other minor code fixes for warnings / deprecations

### CODE REVIEW

**Comment by philsturgeon** [comment]
I've added the id to sites too just like supervisors.

File: Tree Tracker/Models/Api/ProtectEarthSite.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+import CoreLocation
+
+struct ProtectEarthSite: Codable {
+    let id: String
+    let name: String
+    let location: String
+    let url: URL?
+    let plantedTrees: Int
+    // TODO: Do we want to handle coordinates?
```
Comment (philsturgeon) [inline Tree Tracker/Models/Api/ProtectEarthSite.swift:10]: We don't need to worry about site coordinates for now, unless we're planning to have a map view of sites. I mostly added that so we could do this: https://www.protect.earth/test-sites

File: Unit Tests/ProtectEarthCodableTests.swift [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            "url": "https://www.protect.earth/projects/bourton-on-the-water",
+            "plantedTrees": 710,
+            "coordinates": {
+              "latitude": 51.87351,
+              "longitude": -1.90973
+            }
+          },
+          {
+            "id": "497f6eca-6276-4993-cdac-53cbbbba6f08",
+            "name": "Slapton Bottom",
```
Comment (philsturgeon) [inline Unit Tests/ProtectEarthCodableTests.swift:67]: Bahahahaha

**Comment by philsturgeon** [comment]
Added a new token today, `2|3bLACZmtpsoqVyETRVJk3mPUStQ4XpBYnfGJHuq2`.

**Comment by allanlang** [comment]
@philsturgeon oops - this repository is public, so best not to share secrets here. I suggest rotating this one and then sending me a new one via private channel when you get a chance.

**Comment by philsturgeon** [comment]
ha! It’s staging so no drama, but I’ll rotate all the same.

**Pull request merged**